src/dbnode/storage/series/buffer.go

```diff
@@ -692,9 +692,12 @@ func (b *dbBuffer) FetchBlocksForColdFlush(
 	}
 	if bucket, exists := buckets.writableBucket(ColdWrite); exists {
 		bucket.version = version
-	} else {
-		return nil, fmt.Errorf("writable bucket does not exist with block start %s", start)
 	}
+	// No-op if the writable bucket doesn't exist.
+	// This function should only get called for blocks that we know need to be
+	// cold flushed. However, buckets that get attempted to be cold flushed and
+	// fail need to get cold flushed as well. These kinds of buckets will have
```

Comment:
Dumb clarification question--why is the version non writeable on the retry?

Reply:
Nah not dumb. The "writeable" bucket version is always zero. When we cold flush, we mark the writeable bucket a different version so that we know which buckets have flushed - this version stamp is done in the if block above.

However, when we change the version above, we're not *actually* flushed because we don't write to disk until every series in this shard is complete flushing. Therefore, if we for some reason fail to flush for some other series, this bucket that we just marked as a different version needs to be flushed again the next time we try to flush.